packages/network/src/index.ts

```diff
@@ -66,6 +66,10 @@ export class Network extends Connector {
 
     const urls = this.urlMap[chainId]
 
+    if (!urls) {
+      throw new Error(`No provider specified for chainId ${chainId}`)
```

Comment:
I think we need to move towards a codebase where known errors have known subclasses, eg

```
class NoProviderError extends Error {
  constructor(chainId: number) {
    super(`No provider specified for chainId ${chainId}`)
    this.name = 'NoProviderError'
  }
}
```

Reply:
Can we @zzmp have a design doc for this? I think if we want to start applying this, should be a concrete decision and maybe formal proposal to refactor existing ones so that we maintain consistency. I would like to discuss pros/cons.